src/storage/core.rs

```diff
@@ -885,12 +916,14 @@ where
         self.observer.run();
     }
 
-    /// Mark as deleted entries with matching key
+    /// Delete entries with matching key
     /// # Errors
     /// Fails after any disk IO errors.
-    pub async fn mark_all_as_deleted(&self, key: impl AsRef<K>) -> Result<u64> {
+    pub async fn delete(&self, key: impl AsRef<K>, only_if_presented: bool) -> Result<u64> {
         let mut total = 0;
-        total += self.mark_all_as_deleted_active(key.as_ref()).await?;
+        total += self
+            .mark_all_as_deleted_active(key.as_ref(), only_if_presented)
```

Comment:
`write` lock on `safe` is acquired independently in `mark_all_as_deleted_active` and in `mark_all_as_deleted_closed`. This can lead to a situation where the active blob is replaced in between. I think, it is better to acquire that lock once in `delete` function